youtube_dl/extractor/medaltv.py

```diff
@@ -0,0 +1,123 @@
+# coding: utf-8
+
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+from ..utils import (
+    ExtractorError,
+    try_get,
+    float_or_none,
+    int_or_none
+)
+
+
+class MedalTVIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?medal\.tv/clips/(?P<id>[0-9]+)'
+    _TESTS = [{
+        'url': 'https://medal.tv/clips/34934644/3Is9zyGMoBMr',
+        'md5': '7b07b064331b1cf9e8e5c52a06ae68fa',
+        'info_dict': {
+            'id': '34934644',
+            'ext': 'mp4',
+            'title': 'Quad Cold',
+            'description': 'Medal,https://medal.tv/desktop/',
+            'uploader': 'MowgliSB',
+            'timestamp': 1603165266,
+            'upload_date': '20201020',
+            'uploader_id': 10619174,
+        }
+    }, {
+        'url': 'https://medal.tv/clips/36787208',
+        'md5': 'b6dc76b78195fff0b4f8bf4a33ec2148',
+        'info_dict': {
+            'id': '36787208',
+            'ext': 'mp4',
+            'title': 'u tk me i tk u bigger',
+            'description': 'Medal,https://medal.tv/desktop/',
+            'uploader': 'Mimicc',
+            'timestamp': 1605580939,
+            'upload_date': '20201117',
+            'uploader_id': 5156321,
+        }
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        hydration_data = self._search_regex(r'<script>.*hydrationData\s*=\s*({.+?})\s*</script>', webpage, 'hydration data', default='{}')
+        parsed = self._parse_json(hydration_data, video_id)
+
+        # This is necessary because the id of the author is not known in advance.
+        # Will not raise an issue if no profile can be found as this is optional.
+        author_info = try_get(parsed.get('profiles'), lambda x: list(x.values())[0], dict) or {}
+        author_url = 'https://medal.tv/users/{}'.format(author_info.get('id')) if author_info.get('id') else None
```

Comment:
try put value that are used multiple times in a variable(ex: `author_info.get('id')`).
i think the `{}` format is not supported in python 2.6.

Reply:
Oh yes you are correct, looks like `{}` requires a positional argument for python 2.6 and 3.0. Source (https://stackoverflow.com/questions/19668395/str-format-for-python-2-6-gives-error-where-2-7-does-not)

Fixed both issues as follows:
```python
author_id = author_info.get('id')
author_url = 'https://medal.tv/users/{0}'.format(author_id) if author_id else None
```